packages/sw-precaching/src/lib/revisioned-cache-manager.js

```diff
@@ -142,10 +97,78 @@ class RevisionedCacheManager {
       }
 
       event.waitUntil(
-        installHandler({assetsAndHahes: this._cachedAssets})
+        this._performInstallStep()
+        .then(() => {
+          this._close();
+        })
       );
     });
   }
+
+  _performInstallStep() {
+    const assetsAndHahes = this._cachedAssets;
+    const cacheId = null;
+    if (!Array.isArray(assetsAndHahes)) {
+      throw ErrorFactory.createError('assets-not-an-array');
+    }
+
+    const cacheName = cacheId || defaultCacheName;
+
+    return caches.open(cacheName)
+    .then((openCache) => {
+      const cachePromises = assetsAndHahes.map((assetAndHash) => {
+        return this._getRevisionDetails(assetAndHash.path)
+        .then((previousRevisionDetails) => {
+          if (previousRevisionDetails) {
+            if (previousRevisionDetails === assetAndHash.revision) {
+              /* eslint-disable no-console */
```

Comment:
Disabling disallowing console (http://eslint.org/docs/rules/no-console) here. Was this a debug statement we forgot to drop or intentional? :)